### PR TITLE
[IVY] Fix `<ng-template>` and *-attribute parsing

### PR DESCRIPTION
**Pull request by vicb**
2 issues fixed in this PR:
- in r3_template_transform: with `'<div *ngIf="expr as local"></div>'` `ngIf` was reported as both a TextAttribute and a BoundProperty
- in the template compiler, bound property on `<ng-template>` were not taken into account to match directive. That is `<ng-template [ngIf]=cond>` would not trigger the `NgIf` directive.

### CODE REVIEW

File: packages/compiler-cli/test/compliance/r3_view_compiler_template_spec.ts [hunk 1]
```diff
@@ -318,4 +318,50 @@ describe('compiler compliance: template', () => {
...
+              export class MyComponent {}
+
+              @NgModule({declarations: [MyComponent], imports: [CommonModule]})
+              export class MyModule {}
+          `
+      }
+    };
+
+    const template = `
+      const $c0$ = ["attr", "", "boundAttr", ""];
```
Comment (mhevery) [inline packages/compiler-cli/test/compliance/r3_view_compiler_template_spec.ts]: why is the order of attributes reversed with respect to declaration?

File: packages/compiler-cli/test/compliance/r3_view_compiler_template_spec.ts [hunk 2]
```diff
@@ -318,4 +318,50 @@ describe('compiler compliance: template', () => {
...
+        if (rf & 1) {
+          $i0$.ɵT(0, " some-content ");
+        }
+      }
+
+      // ...
+
+      template: function MyComponent_Template(rf, ctx) {
+        if (rf & 1) {
+          $i0$.ɵC(0, Template_0, null, $_c0$);
```
Comment (mhevery) [inline packages/compiler-cli/test/compliance/r3_view_compiler_template_spec.ts]: Why extra `_` in `$_c0$`

File: packages/compiler-cli/test/compliance/r3_view_compiler_template_spec.ts [hunk 3]
```diff
@@ -318,4 +318,50 @@ describe('compiler compliance: template', () => {
...
+        }
+        if (rf & 2) {
+          $i0$.ɵp(0, "boundAttr", $i0$.ɵb(ctx.b));
+        }
+      }`;
+
+    const result = compile(files, angularFiles);
+
+    expectEmit(result.source, template, 'Incorrect template');
```
Comment (mhevery) [inline packages/compiler-cli/test/compliance/r3_view_compiler_template_spec.ts]: Why is it incorrect template?

File: packages/compiler-cli/test/compliance/r3_view_compiler_template_spec.ts (hunk 1 above)
Comment (vicb) [reply packages/compiler-cli/test/compliance/r3_view_compiler_template_spec.ts]: it's attributes first and bound properties next - does not really matters as it is used only for matching directives

File: packages/compiler-cli/test/compliance/r3_view_compiler_template_spec.ts (hunk 3 above)
Comment (vicb) [reply packages/compiler-cli/test/compliance/r3_view_compiler_template_spec.ts]: That's the error message that is displayed when `expectEmit(...)` fails

File: packages/compiler-cli/test/compliance/r3_view_compiler_template_spec.ts (hunk 1 above)
Comment (mhevery) [reply packages/compiler-cli/test/compliance/r3_view_compiler_template_spec.ts]: I thought we generate `["attr", "", AttributeMarker.SelectOnly, "boundAttr"]`

Comment (vicb) [reply packages/compiler-cli/test/compliance/r3_view_compiler_template_spec.ts]: 1) This is not been implemented on the compiler side. I'm wondering if it's still no @pkozlowski-opensource radar ?

2) Probably would be `[AttributeMarker.SelectOnly,  "attr", "boundAttr"]` here as we are visiting a template and attribute are used only to match directives

Comment (mhevery) [reply packages/compiler-cli/test/compliance/r3_view_compiler_template_spec.ts]: I see I captured it here: https://angular-team.atlassian.net/browse/FW-276

**Review by mhevery — APPROVED** [review]

**Comment by vicb** [comment]
http://test/OCL:207216145:BASE:207216165:1533270485936:9972c3dc

**Pull request closed without merging**